### PR TITLE
Supporting confidence calculation using RefuelLLM

### PR DESCRIPTION
**Pull request by DhruvaBansal00**
Screenshot 2023-11-22 at 1 55 11 PM

Confidence calculation will now call https://llm.refuel.ai/models/refuelllm/confidence

### CODE REVIEW

**Comment by DhruvaBansal00** [comment]
Quick summary of changes: we have a new method called `construct_refuel_prompt` in base task which will set the right task properties and call `construct_prompt` to get the refuel llm prompt. We call this right before making a call to the confidence api, add it to the LLMAnnotation object and then compute confidence using this prompt.

**Comment by DhruvaBansal00** [comment]
Screenshot 2023-11-22 at 2 13 30 PM

Working for QA as well now

File: src/autolabel/tasks/base.py [hunk 1]
```diff
@@ -94,6 +91,22 @@ def _is_few_shot_mode(self) -> bool:
...
+        self.use_refuel_prompt_schema = True
+        self.prompt_template = PromptTemplate(
+            input_variables=get_format_variables(
+                self.FEW_SHOT_TEMPLATE_REFUEL_LLM
+                if self._is_few_shot_mode()
+                else self.ZERO_SHOT_TEMPLATE_REFUEL_LLM
+            ),
+            template=self.example_template,
+        )
+        refuel_prompt = self.construct_prompt(input, examples)
```
Comment (nihit) [inline src/autolabel/tasks/base.py:104]: hmm, i suggest instead passing this flag as a parameter to ` self.construct_prompt(...)` to denote which "prompt modes" to construct - the default one or the one customized for refuel llm. that seems cleaner than setting some isntance variavle, and then resetting it again.

**Review by nihit — APPROVED** [review]
lgtm otherwise

**Comment by DhruvaBansal00** [comment]
@nihit re-requesting review based on our discussion. Have tested this implementation with qa and classification tasks as well.

**Review by nihit — APPROVED** [review]

**Pull request merged**